### ISSUE TITLE
@BoundSetter doesn't work for classes

### ISSUE THREAD

**Issue by pdex** [peichhorn/lombok-pg#40]
I'm not sure if I'm using this correctly, but 

``` java
@Data
@BoundPropertySupport
@BoundSetter
public class Animal {
  private String name;
}
```

doesn't cause and property change events to fire, however

``` java
@Data
@BoundPropertySupport
public class Animal {
@BoundSetter
  private String name;
}
```

does cause property change events. I'm using lombok-pg-0.10.1-EDGE.jar.

**Comment by peichhorn** [comment]
The issue here is, that you are using `@Data` together with `@BoundSetter`.
So `@Data` generates the plain old setters and when `@BoundSetter` gets handled, the handler notices that there is already a setter method for the property _name_ and does absolutely nothing. To clarify even further, `@Data` is just a shorthand for: `@RequiredArgsConstructor`, `@Getter`, `@Setter`, `@EqualsAndHashCode` and `@ToString`.

Another important thing to know is, that your second example should only work for eclipse or javac, but not for both, since as far as I remember lombok has an inconsistency there. Since I'm only on a mobile device I can't verify this.

So in short, for the time being mixing `@Data` with any other `@...Getter` or `@...Setter` is a bad idea.

But if you really want to to mix `@Data` and `@BoundSetter` on the type level, I probably could improve each `@...Getter` or `@...Setter` handler in the way that they can manipulate the setter methods lombok has already generated.
Basically if you really want or need it, just give the world and I will see what I can do.

**Comment by pdex** [comment]
Thanks for the swift reply, I was wondering how well this played with `@Data`. I think my use case might be closer to:

``` java
@BoundPropertySupport
@Getter
@BoundSetter
public class Animal {
  private String name;
}
```
